conceptnet5/vectors/query.py

```diff
@@ -244,3 +236,39 @@ def get_similarity(self, query1, query2):
         vec1 = self.get_vector(query1)
         vec2 = self.get_vector(query2)
         return cosine_similarity(vec1, vec2)
+
+    def get_index(self, prefix):
+        """
+        Get an index of a prefix. Using index_cache to store previously retrieved indices
+        instead of retrieving them from the index every time cuts the computation time by 78%.
+        This is particularly useful for Semeval data, where the same words are reused across
+        different test sets.
+
+        If index_cache reaches 50,000 entries, drop it to prevent it from becoming too large.
+        """
+        try:
+            index = self.index_cache[prefix]
+        except KeyError:
+            try:
+                index = self.frame.index.get_loc(prefix, method='bfill')
+                self.index_cache[prefix] = index
+            except KeyError:
+                index = len(self.frame.index)
+                self.index_cache[prefix] = index
+
+        # Drop cache if it contains more than 50000 entries
+        if len(self.index_cache) > 50000:
+            self.index_cache = {}
```

Comment:
This will free up the memory in a kind of roundabout way -- assuming nothing else has a reference to `self.index_cache`, then Python's garbage collector will notice that the old `self.index_cache` isn't being used anymore, and free up the memory used by it and all its contents.

It may not make a difference if garbage collection does its job, but it seems more straightforward to leave the reference the same and clear out its contents, with `self.index_cache.clear()`.

Also, while it's reasonable to assume that nothing else is using the reference to `self.index_cache`, it would be good to follow the Python convention about private fields by changing its name to `self._index_cache`.